addons/account/models/partner.py

```diff
@@ -91,6 +85,38 @@ def _onchange_country_group_id(self):
             self.zip_from = self.zip_to = self.country_id = False
             self.state_ids = [(5,)]
 
+    @api.multi
+    def _convert_zip_values(self):
+        if self.env.context.get('zip_number_converted'):
+            return
+        for record in self:
+            zip_from, zip_to = record.zip_from or False, record.zip_to or False
+            max_length = max(zip_from and len(zip_from) or 0, zip_to and len(zip_to) or 0)
+            if zip_from and zip_from.isdigit():
+                zip_from = zip_from.rjust(max_length, '0')
+            if zip_to and zip_to.isdigit():
+                zip_to = zip_to.rjust(max_length, '0')
+            if zip_from > zip_to:
+                raise ValidationError(_('Invalid "Zip Range", please configure it properly.'))
+            record.with_context({'zip_number_converted': True}).write({
+                'zip_from': zip_from,
+                'zip_to': zip_to
+            })
+        return
+
+    @api.model
+    def create(self, vals):
```

Comment:
@rja-odoo , do not change the value of zip_from and zip_to on create/write method. we just need to validate the value of zip_from and zip_to in constraints.

so move stuff of _convert_zip_values into _check_zip method.

and @api.one is deprecated . so remove it from _check_zip method

Reply:
@rja-odoo , wait

yes we need to change zip values on create/write method. 

In database, zip_from = "4" and zip_to = "10" if zipcode = "05" or zipcode = "5",  result will be False but it should be true

but In database, zip_from = "04" and zip_to = "10". zipcode = "05", result will be seem ok. but zipcode="5",  result will not ok.

@qdp-odoo , can we change zip values on create/write methods ? "4" => "04"